dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+    val output = publishedCET.outputs.last
+
+    val (extCetPrivKey, isWin) =
+      if (publishedCET.txIdBE == dlcSetup.cetWinRemoteTxid) {
+        (cetWinPrivKey, false)
+      } else {
+        (cetLosePrivKey, true)
+      }
+
+    val cetPrivKeyToRemote = extCetPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+    val spendingInfo = P2WPKHV0SpendingInfo(
+      outPoint = TransactionOutPoint(publishedCET.txIdBE, UInt32.one),
+      amount = output.value,
+      scriptPubKey = output.scriptPubKey.asInstanceOf[P2WPKHWitnessSPKV0],
+      signer = cetPrivKeyToRemote,
+      hashType = HashType.sigHashAll,
+      scriptWitness = P2WPKHWitnessV0(cetPrivKeyToRemote.publicKey)
+    )
+
+    val txF =
+      constructClosingTx(privKey = finalPrivKey,
+                         spendingInfo = spendingInfo,
+                         isWin = isWin,
+                         spendsToLocal = false)
+
+    txF.map { tx =>
+      DLCOutcome(
+        fundingTx = dlcSetup.fundingTx,
+        cet = publishedCET,
+        closingTx = tx,
+        cetSpendingInfo = spendingInfo
+      )
+    }
+  }
+
+  /** Constructs and executes on the justice spending branch of a DLC
+    * where a published CET has timed out.
+    *
+    * @return Each transaction published and its spending info
+    */
+  def executeJusticeDLC(
+      dlcSetup: SetupDLC,
+      timedOutCET: Transaction): Future[DLCOutcome] = {
+    val justiceOutput = timedOutCET.outputs.head
+
+    val (extCetPrivKey, cetScriptWitness, isWin) =
+      if (timedOutCET.txIdBE == dlcSetup.cetWinRemoteTxid) {
+        (cetWinPrivKey, dlcSetup.cetWinRemoteWitness, false)
+      } else {
+        (cetLosePrivKey, dlcSetup.cetLoseRemoteWitness, true)
+      }
+
+    val cetPrivKeyJustice = extCetPrivKey.deriveChildPrivKey(UInt32.one).key
+
+    val justiceSpendingInfo = P2WSHV0SpendingInfoFull(
+      outPoint = TransactionOutPoint(timedOutCET.txIdBE, UInt32.zero),
+      amount = justiceOutput.value,
+      scriptPubKey = justiceOutput.scriptPubKey.asInstanceOf[P2WSHWitnessSPKV0],
+      signersWithPossibleExtra = Vector(cetPrivKeyJustice),
+      hashType = HashType.sigHashAll,
+      scriptWitness = cetScriptWitness,
+      conditionalPath = ConditionalPath.nonNestedFalse
+    )
+
+    val justiceSpendingTxF =
+      constructClosingTx(privKey = finalPrivKey,
+                         spendingInfo = justiceSpendingInfo,
+                         isWin = isWin,
+                         spendsToLocal = false)
+
+    justiceSpendingTxF.map { justiceSpendingTx =>
+      DLCOutcome(
+        fundingTx = dlcSetup.fundingTx,
+        cet = timedOutCET,
+        closingTx = justiceSpendingTx,
+        cetSpendingInfo = justiceSpendingInfo
+      )
+    }
+  }
+
+  /** Constructs and executes on the refund spending branch of a DLC
+    *
+    * @return Each transaction published and its spending info
+    */
+  def executeRefundDLC(dlcSetup: SetupDLC): Future[DLCOutcome] = {
```

Comment:
important change, both parties must now call this method separately to get each of their closing txs

Reply:
Link to the spot in dlc specs where this is defined